### PR TITLE
Generate s3 console links for expiring logs

### PR DESCRIPTION
**Pull request by cat5inthecradle**
The presigned link is super convenient for immediate access, but after 72 hours the link is expired and you have to parse the url and hunt for the log file in S3.

image

image

This change adds a permalink to the S3 console. You will have to already be logged in, but this saves steps and mental overhead.

## Follow-up work

Similar functionality is used elsewhere, somewhat inconsistently. We may wish to include this in these other locations:
* https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/lib/contact_rollups_v2.rb#L273
* https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/lib/expired_deleted_account_purger.rb#L221
* https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/ui/runner.rb#L289

## Security

No new information is exposed in these links, and using them requires AWS Authentication.

## PR Checklist:

- [x] Tests provide adequate coverage
- [x] Privacy and Security impacts have been assessed
- [x] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [x] Relevant documentation has been added or updated
- [x] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [x] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: lib/cdo/aws/s3.rb [hunk 1]
```diff
@@ -292,6 +292,25 @@ def self.presigned_url(method, bucket, key, params = {})
...
     end
 
+    # Returns a link to the S3 web console for a given presigned S3 URL.
+    # This is useful for finding a file once the presigned URL has expired.
+    # The user will need to authenticate when using the link.
+    # @param [String] a presigned S3 URL
+    # @return [String] a direct link to the file in the S3 console
+    # @raise [Exception] if the provided link isn't a presigned S3 URL
+    def self.get_console_link_from_presigned(presigned_url)
+      presidned_regex = /^https\:\/\/([a-z0-9][a-z0-9-]{1,61}[a-z0-9])\.s3\.amazonaws.com\/(.*)\?.*$/
```
Comment (davidsbailey) [inline lib/cdo/aws/s3.rb]: nit: typo on presidned

**Review by davidsbailey — APPROVED** [review]
awesome, thanks for jumping on this Darin! makes me look forward to my next dotd shift =)

**Review by Hamms — APPROVED** [review]
Seconded; I'm looking forward to seeing this in action!

**Comment by sureshc** [comment]
What if we modified `AWS::S3::LogUploader` to return both a signed AND a Console Permalink to the uploaded file? Seems odd for us to parse the signed URL to construct the console link.

**Comment by megcrenshaw** [comment]
Thanks for doing this, Darin!

**Comment by cat5inthecradle** [comment]
Suresh and I discussed these various logging use cases, and are strongly leaning towards switching to Cloudwatch Log Streams. With these we could likely have one-click dev access that still requires authentication. The logs would also be more easily queryable.

That said, I'm merging this now as it will provide value now, and be easily cleaned up if we switch to CloudWatch Logs

**Pull request merged**